src/fidesops/service/connectors/query_config.py

```diff
@@ -178,7 +182,7 @@ def update_value_map(
                         f"Unable to generate a query for field {rule_field_path.string_path}: data_type is either not present on the field or not supported for the {strategy_config['strategy']} masking strategy. Received data type: {masking_override.data_type_converter.name}"
                     )
                     continue
-                val: Any = row[rule_field_path.string_path]
+                val: Any = rule_field_path.retrieve_from(row)
```

Comment:
This adjustment lets us get a potentially nested value that was previously retrieved so we can mask it.